warehouse/api/integrity.py

```diff
@@ -0,0 +1,80 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pyramid.httpexceptions import HTTPForbidden, HTTPNotAcceptable, HTTPNotFound
+from pyramid.request import Request
+from pyramid.view import view_config
+
+from warehouse.admin.flags import AdminFlagValue
+from warehouse.cache.http import add_vary, cache_control
+from warehouse.cache.origin import origin_cache
+from warehouse.packaging.models import File
+from warehouse.utils.cors import _CORS_HEADERS
+
+MIME_TEXT_HTML = "text/html"
+MIME_PYPI_INTEGRITY_V1_HTML = "application/vnd.pypi.integrity.v1+html"
+MIME_PYPI_INTEGRITY_V1_JSON = "application/vnd.pypi.integrity.v1+json"
+
+
+def _select_content_type(request: Request) -> str:
+    offers = request.accept.acceptable_offers(
```

Comment:
thought: unrelated to this change, I wonder if this pattern is relatively easy to replace with our use of deprecated `.best_match`